File: GetDressed/src/main/java/getdressed/domain/Order.java
package getdressed.domain;

import getdressed.domain.enums.Status;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "orders")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String fullName;

    private String email;

    private String phone;

    private String zipcode;

    private String address;

    @ManyToOne
    private User user;

    @Enumerated(EnumType.STRING)
    private Status status;

}

File: GetDressed-FE/src/app/modals/order.ts
import { OrderItem } from "./order-item";

export interface Order {
    id?: number;
    fullName?: string;
    email?: string;
    phone?: string;
    zipcode?: string;
    address?: string;
    status?: string;
    orderItems?: OrderItem[];
}

export class COrder implements Order{
    constructor( 
        public id?: number,
        public fullName?: string,
        public email?: string,
        public phone?: string,
        public zipcode?: string,
        public address?: string,
        public status?: string,
        public orderItems?: OrderItem[]
    ){}
}